File: NonsPlayer.Core/Models/Album.cs
using System.Text.Json.Serialization;
using NonsPlayer.Core.Contracts.Models;

namespace NonsPlayer.Core.Models;

public class Album : INonsModel
{
    [JsonPropertyName("create_date")] public DateTime CreateDate;
    [JsonPropertyName("description")] public string Description;
    [JsonPropertyName("musics")] public List<Music> Musics { get; set; }
    [JsonPropertyName("artists")] public List<Artist> Artists { get; set; }
    [JsonIgnore] public string ArtistsName => string.Join("/", Artists.Select(x => x.Name));
    [JsonPropertyName("collection_count")] public int CollectionCount { set; get; }
    [JsonPropertyName("track_count")] public int TrackCount { get; set; }
}

File: src/src/lib/types/album.ts
import type { SongModel } from "./base";
import type { Song } from "./song";
import type { Artist } from "./artist";

export interface Album extends SongModel {
  createDate: string;
  description: string;
  songs: Song[];
  artists: Artist[];
  /** 艺术家名，以 "/" 连接（computed） */
  artistsName: string;
  collectionCount: number;
  trackCount: number;
}
